### PR TITLE
Unifies native and Maven-based standard project wizards

### LINKED ISSUE #929: Prevent entering relative path as project location in new project Maven wizard
Currently, if a relative path is entered as a project location, the path eventually becomes relative to the work directory of Eclipse (i.e., relative to the directory from which Eclipse is launched). This results in creating directories relative to the Eclipse work directory.

This applies to ~~both non-Maven and Maven project wizards~~ the Maven wizard.

### LINKED ISSUE #1168: New Project Wizard for Maven-based Projects Hangs
On a fresh install of Eclipse Neon on Mac OS X with Java 8, when I follow our published quickstart and attempt to create a new Maven-based project using our Wizard, Eclipse hangs when on the "Select an Archetype" Page.

I've attached the thread dump to the issue. 

[stack.txt](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/files/676238/stack.txt)

### LINKED ISSUE #1169: New Project Wizard for Maven-based Projects asks for adding APIs before selecting Archetypes
When going through the New Project Wizard flow for creating a Maven-based App Engine project, we're first asked to select APIs to add (App Engine, Objectify, etc) before selecting an archetype. This is confusing, as the Guestbook archetype probably requires the App Engine API. What is the effect of not selecting the App Engine API but selecting the guestbook archetype?

### LINKED ISSUE #1171: New project wizard for maven guestbook archetype doesn't add endpoints library
Even though the endpoints radio button is checked in the new project wizard. 

Might be an issue in the guestbook archetype or version of archetype we're using. 

The endpoints library is added to the hello world archetype.

### LINKED ISSUE #1326: Unified New Project Wizard page
For discussion. Post 1.0 what do people think of a single wizard like this one?

Flex/standard
Maven/No Maven

One menu item 

image

Subsequent pages would change depending on the selections the user made.

### LINKED ISSUE #1327: Do we need MappingDiscoveryJob or not?
This todo remains in CreateMavenBasedAppEngineStandardProject. If we do need the  MappingDiscoveryJob remove the todo. If we don't, remove the job. 

   /*
     * invoke the Maven lifecycle mapping discovery job
     *
     * todo: is this step necessary? we know the archetype contents and we handle the
     * lifecycle-mappings and rules
     */
    Job job = new MappingDiscoveryJob(archetypeProjects);
    job.schedule();

### LINKED ISSUE #1523: App Engine API is always added when creating HelloWorld Maven project
Even if I don't select the App Engine API in the Maven project wizard, I see below in `pom.xml`:

```xml
    <dependency>
      <groupId>com.google.appengine</groupId>
      <artifactId>appengine-api-1.0-sdk</artifactId>
      <version>${appengine.version}</version>
    </dependency>
```

### LINKED ISSUE #1730: Maven Central dependency warnings when creating a Maven-based App Engine Standard project
Some warnings in the Error Log view:

- Warning 1: `The POM for com.google.cloud.tools:appengine-maven-plugin:jar:1.0.0 is missing, no dependency information available`

- Warning 2: `Failed to retrieve plugin descriptor for com.google.cloud.tools:appengine-maven-plugin:1.0.0: Plugin com.google.cloud.tools:appengine-maven-plugin:1.0.0 or one of its dependencies could not be resolved: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact com.google.cloud.tools:appengine-maven-plugin:jar:1.0.0 has not been downloaded from it before.`

We occasionally observed and noted these warnings also in Travis build logs, but I don't think we really attempted to understand what these mean.

### LINKED ISSUE #1903: src/test/com/foo/bar/java
When creating an app from the maven hello world archetype, somehow we end up with

src/test/package/java

instead of the expected

src/test/java/package

image

This has happened on Linux in Neon with 1.0.3. We haven't seen this or reporduced it elsewhere yet. Using the archetypes from the command line, creates the expected hierarchy src/test/java/package

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1326.

Closes #1903. Closes #1730. Closes #1523. Closes #1327. Closes #1171. Closes #1169. Closes #1168. Closes #929.

This completely removes the `.newproject.maven` bundle (and its test bundle). The only portion I could salvage was the code to auto-set a package name based on a Maven group ID; everything else has been removed.

The code path and the UI of the wizard will become a lot similar to those of the flex wizard. Projects created by the std wizard and the flex wizard will have a lot of similarities too.

The last time I did a manual testing things were working fine, but I'll do some more manual testing while this is being reviewed.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.standard.ftl [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.standard.ftl]: Very similar to the flex `pom.xml` template. I'll highlight the difference below:

```diff
--- ./plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.flex.ftl  2017-06-05 17:34:41.314955620 -0400
+++ ./plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.standard.ftl      2017-06-12 11:44:09.308008253 -0400
@@ -14,8 +14,8 @@
     <appengine.maven.plugin.version>1.3.1</appengine.maven.plugin.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.7</maven.compiler.source>
+    <maven.compiler.target>1.7</maven.compiler.target>
     <maven.compiler.showDeprecation>true</maven.compiler.showDeprecation>
   </properties>
 
@@ -27,8 +27,8 @@
     <!-- Compile/runtime dependencies -->
     <dependency>
       <groupId>javax.servlet</groupId>
-      <artifactId>javax.servlet-api</artifactId>
-      <version>3.1.0</version>
+      <artifactId>servlet-api</artifactId>
+      <version>2.5</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
```

File: plugins/com.google.cloud.tools.eclipse.appengine.ui/plugin.xml [hunk 2]
```diff
@@ -58,18 +58,6 @@
                  style="push">
               <parameter
                     name="newWizardId"
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.ui/plugin.xml]: Removing the Maven-based std wizard (along with the separator between the std wizards and the flex wizard) from the GCP toolbar menu.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProjectTest.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
+/*
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProjectTest.java]: I created an abstract base class `CreateAppEngineWtpProjectTest` for `CreateAppEngineStandardWtpProjectTest` and `CreateAppEngineFlexWtpProjectTest`, which provides common code and tests that can apply to the both tests. Most of the code was simply moved from the two subclasses for code sharing.

**Comment by chanseokoh** [comment]
Closes #1903. Closes #1523. Closes #1327. Closes #1171. Closes #1169. Closes #1168. Closes #929.

**Comment by chanseokoh** [comment]
> Is it possible to split this into two PRs? First add Maven to the regular wizard, then remove the archetype based wizard?

If we're good with an interim state, I can try. (I expect most complications would come from tests.)

**Comment by chanseokoh** [comment]
Actually, I can work on a feature branch for splitting the PR. No interim state.

**Comment by chanseokoh** [comment]
I was referring to a state where the current Maven-based wizard exists but is obsolete due to the new unified wizard checked in. You're suggesting this is OK, so let me do so.

**Comment by chanseokoh** [comment]
Now that #2054 has been checked in, this PR removes the `.newproject.maven` bundle and its test bundle. Most of the changes are file deletion in the bundles, but there do exist small changes outside these bundles.

Ready for review.

**Comment by chanseokoh** [comment]
There is a build error on Mars saying that AERI v.2.0.4 is not in the Mars repo. We haven't changed anything on our side regarding this, so hopefully this is a transient error. However, this is happening only on Mars builds, so I got a bit of bad feeling that this won't work going forward.

```
[ERROR] Failed to resolve target definition /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/eclipse/mars/gcp-eclipse-mars.target: Could not find "org.eclipse.epp.logging.aeri.feature.feature.group/2.0.4.v20170307-1435" in the repositories of the current location -> [Help 1]
```

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/lifecycle-mapping-metadata.xml [hunk 4]
```diff
@@ -1,19 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<lifecycleMappingMetadata>
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/lifecycle-mapping-metadata.xml]: Ah great, versions-maven-plugin 2.3 added their own lifecycle-mapping.  As it should be.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/standard/AppEngineStandardProjectWizard.java [hunk 5]
```diff
@@ -48,7 +48,7 @@ public AppEngineWizardPage createWizardPage() {
     AnalyticsPingManager.getInstance().sendPing(
         AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD,
         AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE,
-        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_STANDARD_NATIVE, getShell());
+        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_STANDARD, getShell());
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/standard/AppEngineStandardProjectWizard.java]: So do we have a way to get native vs maven creation?

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/standard/AppEngineStandardProjectWizard.java]: Not at the moment. For now, we just log a wizard type.

**Pull request merged**